guides-markdown/crowdfund-tx.md

```diff
@@ -0,0 +1,487 @@
+
+In our example, we know there are two inputs. In a more complex application, you might put a cap of say 5, then estimate the fee based on that. If there turn out to be fewer then you just have a relatively high fee.
+
+So, let's next add a utility for estimating a max fee. The way this will work is by creating a mock transaction that uses one of our coins to add inputs up to our maximum, calculate the size of the transaction and the fee based on that amount. We'll also build a utility that we'll need again later for adding coins to a tx and signing the input created.
+
+```javascript
+const getMaxFee = function getMaxFee(maxInputs, coin, address, keyring, rate) {
+  const fundingTarget = 100000000; // 1 BTC (arbitrary for purposes of this function)
+  const testMTX = new MTX();
+
+  testMTX.addOutput({ value: fundingTarget, address })
+
+  while(testMTX.inputs.length < maxInputs) {
+    const index = testMTX.inputs.length;
+    addInput(coin, index, testMTX, keyring);
+  }
+
+  return testMTX.getMinFee(null, rate);
+}
+
+const addInput = function addInput(coin, inputIndex, mtx, keyring, hashType) {
+  const sampleCoin = coin instanceof Coin ? coin : Coin.fromJSON(coin);
+  if(!hashType) hashType = Script.hashType.ANYONECANPAY | Script.hashType.ALL;
+
+  mtx.addCoin(sampleCoin);
+  mtx.scriptInput(inputIndex, sampleCoin, keyring);
+  mtx.signInput(inputIndex, sampleCoin, keyring, hashType);
+  assert(mtx.isSigned(), 'Input was not signed properly');
+}
+
+```
+
+Now we can get the fee for our funder transaction. We'll assume a max of 2 inputs, but this can be variable.
+
+```javascript
+const composeCrowdfund = async function composeCrowdfund() {
+  //...
+  const maxInputs = 2;
+  const maxFee = getMaxFee(
+    maxInputs,
+    funderCoins['0'][0],
+    fundeeAddress,
+    funder1Keyring,
+    txRate
+  );
+
+  console.log(`Based on a rate of ${txRate} satoshis/kb and a tx with max ${maxInputs}`);
+  console.log(`the tx fee should be ${maxFee} satoshis`);
+};
+```
+This should log something like:
+> Based on a rate of 10000 satoshis/kb and a tx with max 2 inputs, the tx fee should be 3380 satoshis
+
+#### Step 6: Construct the Transaction
+Now that we've got our tools and coins ready, we can start to build the transaction! Note that we are going to use the maxFee we calculated earlier and subtract it from the output we add to the mtx.
+
+```javascript
+const composeCrowdfund = async function composeCrowdfund() {
+  //...
+
+  const fundMe = new MTX();
+
+  // add an output with the target funding amount minus the fee calculated earlier
+  fundMe.addOutput({ value: fundingTarget - maxFee, address: fundeeAddress });
+
+  // fund with first funder (using the utility we built earlier)
+  let fundingCoin = funderCoins['0'][0];
```

Comment:
You would also need to manually send rawtransaction to another person, right ? (It would not get in the mempool), in case they are on two different devices.

Reply:
yes for sure. This is really just about creating the transactions. With the wallet system in the next version you're actually sending the transaction.